network/src/main/java/com/renovavision/thecocktaildb/network/data/repo/CocktailsRepo.kt

```diff
@@ -0,0 +1,36 @@
+package com.renovavision.thecocktaildb.network.data.repo
+
+import com.renovavision.thecocktaildb.network.CocktailsApi
+import com.renovavision.thecocktaildb.network.DrinksByQuery.Drink
+import com.renovavision.thecocktaildb.network.DrinksCategory.Category
+import com.renovavision.thecocktaildb.network.DrinksIngredient.Ingredient
+import com.renovavision.thecocktaildb.network.data.Data
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CocktailsRepo @Inject constructor(private val cocktailsApi: CocktailsApi) {
+
+    suspend fun loadData(data: Data<*>) = when (data) {
```

Comment:
@AlexGolovach It's not a case for generic function. `CocktailsRepo` should provide 3 different methods here

Reply:
Fixed